### ISSUE TITLE
Have a default URL at Collections level

### ISSUE THREAD

**Issue by garposmack** [usebruno/bruno#425]
I would like to propose the addition of a feature that allows users to define a default URL, either a variable or a typed URL, at the collection level and or folder level in order to use it on all requests within that collection.

**Comment by ThenTech** [comment]
Definitely a needed feature. Appears to be on the [roadmap](https://github.com/usebruno/bruno/discussions/384) for v1 already.

**Comment by jeremyjumeau** [comment]
Totally agree!

This might also fix response previews with relative urls, for assets or whatever non-absolute url.

Also, this would allow to be less verbose in the path definition:
`/api/foo`
instead of (something like)
`{{baseUrl}}/api/foo`

PS: @helloanoop awesome job and philosphy on Bruno :raised_hands: , sorry not be confident enough in JS to submit a PR

**Comment by peteruithoven** [comment]
It looks like there is a Base URL preset we can set per collection now:
![Screenshot from 2023-11-24 00 45 03@2x](https://github.com/usebruno/bruno/assets/523210/76e123db-e4dd-4dcc-a3cb-98bc73d7be41)

But it doesn't seem to be picked up in the requests within the collection. 
I tried adding a `{{baseUrl}}` variable in the request path, but that also doesn't work. 

Am I missing something?

**Comment by peteruithoven** [comment]
For now I just added a variable to the environment. Useful anyway since, in my case, it actually differers for test and production envs.

**Comment by David-Lannion** [comment]
This issue seems quite related with #292

**Comment by RobertKlohr** [comment]
Maybe a better approach to this request would be to implement variables at the collection level which would align with the issue @David-Lannion mentioned in the previous comment.

**Comment by Ativerc** [comment]
Ran into this bug today. The collection's preset Base URL is not being picked up. 

I'm a dummy...

Now I understand what @peteruithoven meant when he said "Update: Ah wait, it's just pre-filled when you create a new request..." 

It just prefills the input textbox with the baseURL when you create the new request. I was wondering why the baseURL was there. Maybe it copied it from my clipboard? Nah, it is just copying it from the preset baseURL for the collection.

Definitely needs an UX improvement.

**Comment by unfortunately-react** [comment]
I added a base url, but all the imported requests still only have the request path. Would like to not have to manually update every request.

**Comment by sanjai0py** [comment]
The feature to add a default base URL at the collection level has already been implemented in Bruno. Thank you, @garposmack, for submitting this valuable enhancement request. 

> It definitely needs a UI/UX improvement.

I agree, @Ativerc. Could you please create a new issue for this and include any additional suggestions or ideas you have?

**Comment by bryant-ferguson** [comment]
Can someone point to documentation on how this is supposed to work or explain it. I have imported my OpenAPI spec and set Base URL to my local test instance: "http://localhost:8000", but clearly I'm misunderstanding something because when I select an API endpoint to execute, it does not insert the provided Base URL at the front of that request.
![Screenshot 2024-10-24 175847](https://github.com/user-attachments/assets/154ef9a6-0ea5-4de9-90c7-cfb1b6a31165)
![Screenshot 2024-10-24 175854](https://github.com/user-attachments/assets/a722e060-2dfd-4b2a-8a0f-7c16c08ef884)

**Comment by bryant-ferguson** [comment]
Okay, for anyone new to Bruno and not understanding what those above were saying about "Base URL". It is pre-filled when creating a new "request", that is, you're creating a brand new verb and path option on the left-hand side. So, it's not useful for a pre-existing API as far as I can tell. See here, my "Base URL" was set as above and it is filled into this dialog box...
![Screenshot 2024-10-24 182037](https://github.com/user-attachments/assets/5aeefcd3-f425-4908-946b-754b5b43da1b)

**Comment by w4matt** [comment]
It would also be useful if the base URL for every request changed when the base URL setting is updated. Right now, it looks like the base URL is just a "prefill" for new requests and not actually a variable that can be changed and update all existing requests.

**Comment by bonitor** [comment]
Hello there, 

I just came across this thread while creating my first 🐶 Bruno collection. I know it's old - but I just wanted to contribute by summarizing and showing a final answer that might satisfy the search intent of those finding this page. 

It seems, many people have similar needs with e.g. a **base url for every request in their collection**, but misunderstand how the Collection > Presets > Base URL input works. 

We're lazy 😉: Of course we don't want to type http://localhost:3000 all the time. And we especially don't want to change it to http://localhost:3001, if 2 days later we're working on two projects at once.

## tldr:

1. Create an **Environment** with a **{{base_url}}**. 
2. Set Collection > **Presets > Base URL** to {{base_url}}.
3. New requests are pre-filled with {{base_url}} - just add "/api/path/action" without "http://host.bla:3000".

As mentioned by @Ativerc and @w4matt, the "Base URL" input at the collection level in Bruno only pre-fills the URL field when creating a new request but does not automatically apply to or update existing requests. Many users mistakenly expect it to function as a global variable or auto-prefix, affecting all requests in the collection, which it currently does not. This has caused confusion and calls for UI/UX improvements. But I think we're good!

You can use environments, the collection's presets and variables in requests, to never type your base url again.

## 1. Environment

When working on a machine, you probably fire up the project with the same url:port every time. You can create an environment variable like {{base_url}}.

![Image](https://github.com/user-attachments/assets/392cd294-29ec-4908-8601-511d850d6159)

## 2. Collection's Presets 

You can put this variable in the collection's presets as the Base URL - it will not resolve, just pre-fill the URL field with literally {{base_url}} when creating a new request.

![Image](https://github.com/user-attachments/assets/cbddc08c-4371-4366-ac84-3a0dea3bfe69)

## 3. New requests

When creating a new request, the input is pre-filled with the variable. Just append "/api/path/action" (e.g. "{{base_url}}/api/tasks")

![Image](https://github.com/user-attachments/assets/baae7887-6700-4de7-9065-b8712e30093d)

## 4. Making a request

Bruno even shows, if the variable {{base_url}} is set in your environment, by using green or red text-color. Here i have "forgotten" to select an environment (see dropdown top right).

![Image](https://github.com/user-attachments/assets/35cf944f-c56e-4b73-98bb-0f3d67a48b2a)

Here, I have selected my "Work Laptop Env" environment. 

![Image](https://github.com/user-attachments/assets/ce4c20c3-6209-4329-9656-a8463edfaf80)

I know. Lot's of text. But I hope this quickly helps the next reader. 

👋

**Comment by hong-tm** [comment]
> ## 3. New requests
> When creating a new request, the input is pre-filled with the variable. Just append "/api/path/action" (e.g. "{{base_url}}/api/tasks")
> 
> ![Image](https://github.com/user-attachments/assets/baae7887-6700-4de7-9065-b8712e30093d)
> 

This is the best solution, for those who use `base_url/api/tasks` don't forget to put the double braces `{{ }}` for the environment key 🤗 => `{{base_url}}/api/tasks`

**Comment by mnemotic** [comment]
@bonitor Thank you! 🙏

**Comment by alekwo** [comment]
> Many users **mistakenly expect** it to function as a global variable or auto-prefix, affecting all requests in the collection, which it currently does not.

I don't think this is a mistaken expectation. This is the most logical assumption related to how such feature should behave.
